### PR TITLE
Refactor logging functions to use %s formatting instead of f-strings

### LINKED ISSUE #3820: Comply with logging-fstring-interpolation / W1203
Some logging statements aren't complying with [logging-fstring-interpolation / W1203](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html). 

Examples:
 
https://github.com/mozilla/bugbug/blob/d454491af97ae42256f9e7960c560b1f7e4eca97/bugbug/models/browsername.py#L72-L76

https://github.com/mozilla/bugbug/blob/d454491af97ae42256f9e7960c560b1f7e4eca97/scripts/regressor_finder.py#L342-L344

### PR DESCRIPTION
**Pull request by StarKBhaviN**
Fixes #3820

You can use % formatting but leave interpolation to the logging function by passing the parameters as arguments.
If logging-not-lazy is disabled then you can use % formatting as normal.

### CODE REVIEW

**Review by suhaibmujahid — COMMENTED** [review]
Thank you for the PR, @StarKBhaviN! Could you please fix the failing tests?

Using pre-commit will help in fixing the code style errors: https://github.com/mozilla/bugbug#auto-formatting

**Comment by StarKBhaviN** [comment]
[image: image.png]

Why i am getting this error

File: .flake8
Comment (suhaibmujahid) [inline .flake8:1]: Unrelated change

File: .pre-commit-config.yaml
Comment (suhaibmujahid) [inline .pre-commit-config.yaml:1]: Unrelated changes

File: bugbug/labels/regression_bug_nobug.csv
Comment (suhaibmujahid) [inline bugbug/labels/regression_bug_nobug.csv:1]: Unrelated changes

File: http_service/.dockerignore
Comment (suhaibmujahid) [inline http_service/.dockerignore:1]: Unrelated change

File: scripts/regressor_finder.py [hunk 1]
```diff
@@ -500,7 +503,7 @@ def results():
                     exc = future.exception()
                     if exc is not None:
                         logger.info(
-                            f"Exception {exc} while analyzing {futures[future]}"
+                            "Exception %d while analyzing %s", exc, futures[future]
```
Comment (suhaibmujahid) [inline scripts/regressor_finder.py:506]: The value of `exc` is an Exception, not an integer.

File: scripts/trainer_extract_args.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 logger = logging.getLogger(__name__)
 
 
-def get_model_name() -> str | None:
+def get_model_name() -> str or None:
```
Comment (suhaibmujahid) [inline scripts/trainer_extract_args.py:14]: Unrelated change

File: tests/fixtures/bug_features/nightly_uplift.json
Comment (suhaibmujahid) [inline tests/fixtures/bug_features/nightly_uplift.json:1]: Unrelated change

File: tests/fixtures/bugs.json
Comment (suhaibmujahid) [inline tests/fixtures/bugs.json:1]: Unrelated change

File: bugbug/models/regressor.py [hunk 3]
```diff
@@ -278,7 +278,7 @@ def evaluation(self) -> None:
         total_regressions = sum(1 for _, is_reg in results if is_reg)
         average_regression_rate = total_regressions / total_landings
 
-        logger.info("Average risk is %d", average_regression_rate)
```
Comment (suhaibmujahid) [inline bugbug/models/regressor.py:281]: This is a valid fix but unrelated to the goal of this PR. We could open a new issue and fix it in a separate PR (with similar cases if there are any)

**Review by suhaibmujahid — COMMENTED** [review]
Thank you, @StarKBhaviN! Could you please take care of the following:
- Drop unrelated changes
- Link the PR to the issue that it is fixing (in the PR description[^1])
- Edit the PR title to clarify that this is related to logging  

[^1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

File: .flake8
Comment (StarKBhaviN) [reply .flake8:1]: This change made automatically when running command...

- pre-commit run --all-files

File: .pre-commit-config.yaml
Comment (StarKBhaviN) [reply .pre-commit-config.yaml:1]: This change made automatically when running command...

- pre-commit install

File: bugbug/labels/regression_bug_nobug.csv
Comment (StarKBhaviN) [reply bugbug/labels/regression_bug_nobug.csv:1]: This change made automatically when running command...

- pre-commit run --all-files

File: http_service/.dockerignore
Comment (StarKBhaviN) [reply http_service/.dockerignore:1]: This change made automatically when running command...

- pre-commit run --all-files

File: tests/fixtures/bug_features/nightly_uplift.json
Comment (StarKBhaviN) [reply tests/fixtures/bug_features/nightly_uplift.json:1]: This change made automatically when running command...

- pre-commit run --all-files

File: scripts/regressor_finder.py (hunk 1 above)
Comment (StarKBhaviN) [reply scripts/regressor_finder.py:506]: Yeah fixed that!!!

File: scripts/trainer_extract_args.py (hunk 2 above)
Comment (StarKBhaviN) [reply scripts/trainer_extract_args.py:14]: Yeah fixed it

File: bugbug/models/regressor.py (hunk 3 above)
Comment (StarKBhaviN) [reply bugbug/models/regressor.py:281]: Okay got it

File: .pre-commit-config.yaml
Comment (suhaibmujahid) [reply .pre-commit-config.yaml:1]: It seems that the file was modified by mistake. The `pre-commit install` command should not modify the config file. Modifying the config file caused the other modification.

File: bugbug/labels/regression_bug_nobug.csv
Comment (suhaibmujahid) [reply bugbug/labels/regression_bug_nobug.csv:1]: You do not need to execute `pre-commit run --all-files`; once you have pre-commit installed (i.e., `pre-commit install`) pre-commit will run automatically when you commit. If you want to run it manually, using `pre-commit run` should be enough to lint the modified files.

File: .flake8
Comment (suhaibmujahid) [reply .flake8:1]: This is due to the changes in the config file (i.e., `.pre-commit-config.yaml`). All unrelated changes should be dropped.

**Comment by StarKBhaviN** [comment]
image

This 3 errors does not letting me commit.

Issues resolved :
- Reverted commit eed0ef2e89d57fa9d0c19c5a6a289b51ad8418dc
- Changed only required code.
- Now trying to commit i am getting error. HELP!

**Comment by jpangas** [comment]
@StarKBhaviN , Bugbug requires Python 3.10+ as mentioned in the [Setup and Prerequisities](https://github.com/mozilla/bugbug#setup-and-prerequisites) 

> Currently, Python 3.10+ is required. You can double check the version we use by looking at setup.py.

You're using Python `3.9.13 `which is not supported. I would advise you create a virtual environment using `venv` or `conda` for `bugbug` and its dependencies. Remember to use Python 3.10+

**Comment by StarKBhaviN** [comment]
image
Still getting same error after creating virtual env and reinstalling dependencies

**Comment by jpangas** [comment]
Let's chat in the bugbug matrix room. It will be easier to follow up from there and others will be available to help.

**Comment by suhaibmujahid** [comment]
>  Still getting same error after creating virtual env and reinstalling dependencies

Try to re-install `pre-commit` after activating venv:
```
pre-commit uninstall
pre-commit install
```

**Comment by StarKBhaviN** [comment]
image

bugbug lint is failing in codespell

**Comment by suhaibmujahid** [comment]
> bugbug lint is failing in codespell

Because the PR still changes in the `.pre-commit-config.yaml` file. All unrelated changes should be dropped. Please just keep the changes related to the logging statements mentioned in the issue.

**Review by suhaibmujahid — COMMENTED** [review]
Thank you @StarKBhaviN! The changes to the following files are not related to this PR. Please revert the changes:
- .flake8
- bugbug/labels/regression_bug_nobug.csv
- http_service/.dockerignore
- tests/fixtures/bug_features/nightly_uplift.json
- tests/fixtures/bugs.json

File: bugbug/models/regressor.py (hunk 3 above)
Comment (suhaibmujahid) [reply bugbug/models/regressor.py:281]: I filed a separate issue for this (i.e., #3837). This fix is not related to this PR. Please revert it.

File: scripts/commit_classifier.py [hunk 4]
```diff
@@ -294,7 +294,7 @@ def apply_phab(self, hg, phabricator_deployment, diff_id):
             # Stop as soon as a base revision is available
             if self.has_revision(hg, patch.base_revision):
                 logger.info(
-                    f"Stopping at diff {patch.id} and revision {patch.base_revision}"
+                    "Stopping at diff %f and revision %f", patch.id, patch.base_revision
```
Comment (suhaibmujahid) [inline scripts/commit_classifier.py:297]: Could you clarify why `%f` is used for `patch.id`?

File: scripts/commit_classifier.py [hunk 5]
```diff
@@ -383,7 +383,7 @@ def load_user(phid):
                 message = replace_reviewers(message, reviewers)
 
             logger.info(
-                f"Applying {patch.phid} from revision {revision['id']}: {message}"
+                "Applying %f from revision {revision['id']}: %s", patch.phid, message
```
Comment (suhaibmujahid) [inline scripts/commit_classifier.py:386]: The revision id here is missing.

File: scripts/maintenance_effectiveness_indicator.py [hunk 6]
```diff
@@ -54,7 +54,7 @@ def main() -> None:
     )
 
     for factor, value in result.items():
-        logger.info(f"{factor}: {round(value, 2) if value != math.inf else value}")
+        logger.info("%s : %d", factor, round(value, 2) if value != math.inf else value)
```
Comment (suhaibmujahid) [inline scripts/maintenance_effectiveness_indicator.py:57]: The whitespace was not present in the original login statement:
```suggestion
        logger.info("%s: %d", factor, round(value, 2) if value != math.inf else value)
```

File: scripts/regressor_finder.py [hunk 7]
```diff
@@ -608,11 +617,11 @@ def evaluate(bug_introducing_commits):
             misassigned_regressors += 1
 
     logger.info(
-        f"Perfectly found {perfect_regressors} regressors out of {all_regressors}"
+        "Perfectly found %d regressors out of %d", perfect_regressors, all_regressors
     )
     logger.info("Found %d regressors out of %d", found_regressors, all_regressors)
     logger.info(
-        f"Misassigned {misassigned_regressors} regressors out of {all_regressors}"
+        "Misassigned %d regressors out of %d", {misassigned_regressors}, all_regressors
```
Comment (suhaibmujahid) [inline scripts/regressor_finder.py:624]:
```suggestion
        "Misassigned %d regressors out of %d", misassigned_regressors, all_regressors
```

File: scripts/commit_classifier.py (hunk 4 above)
Comment (StarKBhaviN) [reply scripts/commit_classifier.py:297]: It should not be %f.
I am confused between %d and %s, so now i am using .format method

File: scripts/commit_classifier.py [hunk 8]
```diff
@@ -294,7 +294,9 @@ def apply_phab(self, hg, phabricator_deployment, diff_id):
             # Stop as soon as a base revision is available
             if self.has_revision(hg, patch.base_revision):
                 logger.info(
-                    f"Stopping at diff {patch.id} and revision {patch.base_revision}"
+                    "Stopping at diff {} and revision {}".format(
+                        patch.id, patch.base_revision
+                    )
```
Comment (suhaibmujahid) [inline scripts/commit_classifier.py:299]: The goal of the issue is to move from "f-string" to "%s formatting", not to `str.format`. See the reason here: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html

If you are in doubt about what the type is, you could execute the script to check. If it is hard to determine, `%s` should be okay in most cases.

File: scripts/commit_classifier.py [hunk 9]
```diff
@@ -383,7 +383,7 @@ def load_user(phid):
                 message = replace_reviewers(message, reviewers)
 
             logger.info(
-                f"Applying {patch.phid} from revision {revision['id']}: {message}"
+                "Applying %s from revision %s: %s", patch.phid, message, revision["id"]
```
Comment (suhaibmujahid) [inline scripts/commit_classifier.py:386]: The order here is not correct:
```suggestion
                "Applying %s from revision %s: %s", patch.phid, revision["id"], message
```

**Review by suhaibmujahid — CHANGES_REQUESTED** [review]
Please rename the PR to something more descriptive (e.g., “Use %s formatting instead of f-string in logging functions”). Also, could you please link to the issue in the PR description?[^1]

[^1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

File: scripts/regressor_finder.py [hunk 10]
```diff
@@ -340,7 +344,7 @@ def mercurial_to_git(revs):
             for bug_introducing_commit in prev_bug_introducing_commits
         )
         logger.info(
-            f"Already classified {len(prev_bug_introducing_commits)} commits..."
+            "Already classified %s commits...", len(prev_bug_introducing_commits)
```
Comment (suhaibmujahid) [inline scripts/regressor_finder.py:347]: The value here is an integer:
```suggestion
            "Already classified %d commits...", len(prev_bug_introducing_commits)
```

**Review by suhaibmujahid — APPROVED** [review]
LGTM! Thank you, @StarKBhaviN!
